pkg/controller/monitoring/monitoring_controller.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/controller/installation"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/controller/utils/imageset"
+	"github.com/tigera/operator/pkg/render"
+)
+
+var log = logf.Log.WithName("controller_monitoring")
+
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.EnterpriseCRDExists {
+		return nil
+	}
+	return add(mgr, newReconciler(mgr, opts))
+}
+
+func newReconciler(mgr manager.Manager, opts options.AddOptions) reconcile.Reconciler {
+	r := &ReconcileMonitoring{
+		client:   mgr.GetClient(),
+		scheme:   mgr.GetScheme(),
+		provider: opts.DetectedProvider,
+		status:   status.New(mgr.GetClient(), "monitoring", opts.KubernetesVersion),
+	}
+	r.status.Run()
+	return r
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	c, err := controller.New("monitoring-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create monitoring-controller: %w", err)
+	}
+
+	// watch for primary resource changes
+	if err = c.Watch(&source.Kind{Type: &operatorv1.Monitoring{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("monitoring-controller failed to watch primary resource: %w", err)
+	}
+
+	if err = utils.AddNetworkWatch(c); err != nil {
+		return fmt.Errorf("monitoring-controller failed to watch Network resource: %w", err)
```

Comment:
I assume you copied this block from somewhere. Anyway, no resource called Network exists, it is called Installation.